From 00c67a2060b8e076a17b1d70bf82ddc83f95eb0c Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Wed, 9 Nov 2022 15:34:23 +0100
Subject: [PATCH] Revert "Merge pull request
 https://github.com/archlinux/mkinitcpio/pull/69 from
 WhyNotHugo/merge-libalpm-scripts"

This reverts commit 6e03c7460adee080ce8e2c72f3918eef9b5cee6c, reversing
changes made to a7d99c8240bc53e22858589b3eba1805a7ac27e7.
---
 Makefile                                 |  3 +-
 libalpm/hooks/60-mkinitcpio-remove.hook  |  2 +-
 libalpm/hooks/90-mkinitcpio-install.hook |  2 +-
 libalpm/scripts/mkinitcpio               | 95 ------------------------
 libalpm/scripts/mkinitcpio-install       | 44 +++++++++++
 libalpm/scripts/mkinitcpio-remove        | 61 +++++++++++++++
 6 files changed, 109 insertions(+), 98 deletions(-)
 delete mode 100644 libalpm/scripts/mkinitcpio
 create mode 100644 libalpm/scripts/mkinitcpio-install
 create mode 100644 libalpm/scripts/mkinitcpio-remove

diff --git a/Makefile b/Makefile
index e6f3789..9ef0f6a 100644
--- a/Makefile
+++ b/Makefile
@@ -73,8 +73,9 @@ install: all
 	install -m644 shell/zsh-completion $(DESTDIR)/usr/share/zsh/site-functions/_mkinitcpio
 
 	install -m644 libalpm/hooks/90-mkinitcpio-install.hook $(DESTDIR)/usr/share/libalpm/hooks/90-mkinitcpio-install.hook
+	install -m755 libalpm/scripts/mkinitcpio-install $(DESTDIR)/usr/share/libalpm/scripts/mkinitcpio-install
 	install -m644 libalpm/hooks/60-mkinitcpio-remove.hook $(DESTDIR)/usr/share/libalpm/hooks/60-mkinitcpio-remove.hook
-	install -m755 libalpm/scripts/mkinitcpio $(DESTDIR)/usr/share/libalpm/scripts/mkinitcpio
+	install -m755 libalpm/scripts/mkinitcpio-remove $(DESTDIR)/usr/share/libalpm/scripts/mkinitcpio-remove
 
 doc: $(MANPAGES)
 man/%: man/%.txt Makefile
diff --git a/libalpm/hooks/60-mkinitcpio-remove.hook b/libalpm/hooks/60-mkinitcpio-remove.hook
index 9600ce1..9827afc 100644
--- a/libalpm/hooks/60-mkinitcpio-remove.hook
+++ b/libalpm/hooks/60-mkinitcpio-remove.hook
@@ -12,5 +12,5 @@ Target = mkinitcpio-git
 [Action]
 Description = Removing linux initcpios...
 When = PreTransaction
-Exec = /usr/share/libalpm/scripts/mkinitcpio remove
+Exec = /usr/share/libalpm/scripts/mkinitcpio-remove
 NeedsTargets
diff --git a/libalpm/hooks/90-mkinitcpio-install.hook b/libalpm/hooks/90-mkinitcpio-install.hook
index 5b02fb8..cf480bd 100644
--- a/libalpm/hooks/90-mkinitcpio-install.hook
+++ b/libalpm/hooks/90-mkinitcpio-install.hook
@@ -8,5 +8,5 @@ Target = usr/lib/initcpio/*
 [Action]
 Description = Updating linux initcpios...
 When = PostTransaction
-Exec = /usr/share/libalpm/scripts/mkinitcpio install
+Exec = /usr/share/libalpm/scripts/mkinitcpio-install
 NeedsTargets
diff --git a/libalpm/scripts/mkinitcpio b/libalpm/scripts/mkinitcpio
deleted file mode 100644
index 0a75ee1..0000000
--- a/libalpm/scripts/mkinitcpio
+++ /dev/null
@@ -1,95 +0,0 @@
-#!/usr/bin/env bash
-
-set -e
-
-args=()
-package=0
-
-process_preset() {
-    if [[ -n "$pkgbase" && -e "$preset" ]]; then
-        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
-            if [[ ! -e "$preset.pacsave" ]]; then
-                # save the preset as pacsave
-                mv -- "$preset" "$preset.pacsave" && return 0
-            fi
-        else
-            # remove the preset
-            rm -- "$preset" && return 0
-        fi
-    fi
-}
-
-install_kernel() {
-    preset="/etc/mkinitcpio.d/${1}.preset"
-    if [[ ! -e "$preset" ]]; then
-        if [[ -e "$preset.pacsave" ]]; then
-            # move the pacsave to the template
-            mv -- "${preset}.pacsave" "$preset"
-        else
-            # create the preset from the template
-            sed "s|%PKGBASE%|${1}|g" /usr/share/mkinitcpio/hook.preset \
-                | install -Dm644 /dev/stdin "$preset"
-        fi
-    fi
-
-    # always install the kernel
-    install -Dm644 "$line" "/boot/vmlinuz-${1}"
-
-    # compound args for each kernel
-    args+=(-p "$1")
-}
-
-remove_kernel() {
-    # remove the actual kernel and images for the package being removed
-    kernel="/boot/vmlinuz-${1}"
-    preset="/etc/mkinitcpio.d/${1}.preset"
-    initramfs="/boot/initramfs-${1}.img"
-    fallback_initramfs="/boot/initramfs-${1}-fallback.img"
-    # remove the installed kernel
-    rm -f -- "$kernel"
-
-    process_preset "$1" "$preset"
-
-    # remove images
-    rm -f -- "$initramfs" "$fallback_initramfs"
-}
-
-while read -r line; do
-    if [[ "$line" != */vmlinuz ]]; then
-        # triggers when it's a change to usr/lib/initcpio/*
-        package=1
-        continue
-    fi
-
-    if ! read -r pkgbase > /dev/null 2>&1 < "${line%/vmlinuz}/pkgbase"; then
-        # if the kernel has no pkgbase, we skip it
-        continue
-    fi
-
-    case "$1" in
-        install) install_kernel "$pkgbase";;
-        remove) remove_kernel "$pkgbase";;
-    esac
-done
-
-if (( package )) && compgen -G /etc/mkinitcpio.d/"*.preset" > /dev/null; then
-    case "$1" in
-        install)
-            # change to use all presets
-            args=(-P)
-            ;;
-        remove)
-          shopt -s nullglob
-          for preset in /etc/mkinitcpio.d/*.preset; do
-              pkgbase=${preset##*/}
-              pkgbase=${pkgbase%.preset}
-              process_preset "$pkgbase" "$preset"
-          done
-          shopt -u nullglob
-          ;;
-    esac
-fi
-
-if [[ "$1" == "install" ]] && (( ${#args[@]} )); then
-    mkinitcpio "${args[@]}"
-fi
diff --git a/libalpm/scripts/mkinitcpio-install b/libalpm/scripts/mkinitcpio-install
new file mode 100644
index 0000000..d2e5041
--- /dev/null
+++ b/libalpm/scripts/mkinitcpio-install
@@ -0,0 +1,44 @@
+#!/bin/bash -e
+
+args=()
+all=0
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
+        # triggers when it's a change to usr/lib/initcpio/*
+        all=1
+        continue
+    fi
+
+    if ! read -r pkgbase > /dev/null 2>&1 < "${line%/vmlinuz}/pkgbase"; then
+        # if the kernel has no pkgbase, we skip it
+        continue
+    fi
+
+    preset="/etc/mkinitcpio.d/${pkgbase}.preset"
+    if [[ ! -e $preset ]]; then
+        if [[ -e $preset.pacsave ]]; then
+            # move the pacsave to the template
+            mv "${preset}.pacsave" "$preset"
+        else
+            # create the preset from the template
+            sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset \
+                | install -Dm644 /dev/stdin "$preset"
+        fi
+    fi
+
+    # always install the kernel
+    install -Dm644 "${line}" "/boot/vmlinuz-${pkgbase}"
+
+    # compound args for each kernel
+    args+=(-p "${pkgbase}")
+done
+
+if (( all )) && compgen -G /etc/mkinitcpio.d/"*.preset" > /dev/null; then
+    # change to use all presets
+    args=(-P)
+fi
+
+if (( ${#args[@]} )); then
+    mkinitcpio "${args[@]}"
+fi
diff --git a/libalpm/scripts/mkinitcpio-remove b/libalpm/scripts/mkinitcpio-remove
new file mode 100644
index 0000000..17b4b31
--- /dev/null
+++ b/libalpm/scripts/mkinitcpio-remove
@@ -0,0 +1,61 @@
+#!/bin/bash -e
+
+package=0
+
+process_preset() {
+    if [[ -n "${pkgbase}" && -e $preset ]]; then
+        if ! cmp $preset > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+            if [[ ! -e $preset.pacsave ]]; then
+                # save the preset as pacsave
+                mv $preset $preset.pacsave && return 0
+            fi
+        else
+            # remove the preset
+            rm $preset && return 0
+        fi
+    fi
+}
+
+while read -r line; do
+    if [[ $line != */vmlinuz ]]; then
+        # triggers when it's a change to usr/lib/initcpio/*
+        package=1
+        continue
+    fi
+
+    if ! read -r pkgbase > /dev/null 2>&1 < "${line%/vmlinuz}/pkgbase"; then
+        # if the kernel has no pkgbase, we skip it
+        continue
+    fi
+
+    # remove the actual kernel and images for the package being removed
+    kernel="/boot/vmlinuz-${pkgbase}"
+    preset="/etc/mkinitcpio.d/${pkgbase}.preset"
+    initramfs="/boot/initramfs-${pkgbase}.img"
+    fallback_initramfs="/boot/initramfs-${pkgbase}-fallback.img"
+    if [[ -e $kernel ]]; then
+        # remove the installed kernel
+        rm $kernel
+    fi
+
+    process_preset "${pkgbase}" $preset
+
+    if [[ -e $initramfs ]]; then
+        # remove the main image
+        rm $initramfs
+    fi
+    if [[ -e $fallback_initramfs ]]; then
+        # remove the fallback image
+        rm $fallback_initramfs
+    fi
+done
+
+if (( package )) && compgen -G /etc/mkinitcpio.d/"*.preset" > /dev/null; then
+    shopt -s nullglob
+    for preset in /etc/mkinitcpio.d/*.preset; do
+        pkgbase=${preset##*/}
+        pkgbase=${pkgbase%.preset}
+        process_preset "${pkgbase}" $preset
+    done
+    shopt -u nullglob
+fi
-- 
2.38.1.385.g3b08839926

